src/pages/Login.jsx

```diff
@@ -0,0 +1,153 @@
+import React, { useState, useContext } from 'react';
+import { Redirect } from 'react-router-dom';
+import { makeStyles } from '@material-ui/core/styles';
+import {
+    Avatar,
+    Button,
+    TextField,
+    Grid,
+    Typography,
+    Container
+} from "@material-ui/core";
+import LockOutlinedIcon from '@material-ui/icons/LockOutlined';
+import links from "../constants/links";
+import { jwt } from '../services/authentication';
+import AuthContext from '../contexts/AuthContext';
+
+const useStyles = makeStyles((theme) => ({
+    root: {
+        marginTop: theme.spacing(10),
+    },
+    paper: {
+        marginTop: theme.spacing(8),
+        display: 'flex',
+        flexDirection: 'column',
+        alignItems: 'center',
+    },
+    error: {
+        marginTop: theme.spacing(2),
+        paddingLeft: theme.spacing(1),
+        color: '#eb2f06',
+    },
+    form: {
+        width: '100%',
+    },
+    submit: {
+        height: '3em',
+        margin: theme.spacing(2, 0, 2),
+    },
+}));
+
+const initAccount = {
+    email: '',
+    password: ''
+}
+
+export default () => {
+    const classes = useStyles();
+    const [account, setAccount] = useState(initAccount)
+    const [error, setError] = useState('')
+    const { auth, setAuth } = useContext(AuthContext)
+
+    const handleChange = e => {
+        // TODO: validation
+        setAccount({ ...account, [e.target.name]: e.target.value })
+    }
+
+    const handleSubmit = e => {
+        e.preventDefault();
+        jwt.login(account)

```

Comment:
Does this need a return?

Reply:
Do you mean **jwt.login** method? as you know, it's a wrapper for **axios.post**, so it returns **promise object** and I believe it should end its life cycle here after its fulfilled and rejected handlers sequence. Let me know if I misunderstood your question!! other advice and opinion's also welcomed!!

Reply:
Oh, I understand what you mean, stupid!. As far as I know "return false" for onsubmit handler only works for jQuery event handler and it does nothing for regular one. so I think it's enough with e.preventDefault(). Let me know if I'm wrong and thanks for the good question!